docs/pages/reference/operator-resources/resources.teleport.dev_rolesv7.mdx

```diff
@@ -0,0 +1,402 @@
+description: Provides a comprehensive list of fields in the TeleportRoleV7 resource available through the Teleport Kubernetes operator.
+tocDepth: 3
+---
+
+{/*Auto-generated file. Do not edit.*/}
+{/*To regenerate, navigate to integrations/operator and run "make crd-docs".*/}
+
+This guide is a comprehensive reference to the fields available to configure in
+the `TeleportRoleV7` resource, which you can apply after installing
+the Teleport Kubernetes operator.
+
+
+## resources.teleport.dev/v1
+
+**apiVersion:** resources.teleport.dev/v1
+
+|Field|Type|Description|
+|---|---|---|
+|apiVersion|string|APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources|
+|kind|string|Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds|
+|metadata|object||
+|spec|[object](#spec)|Role resource definition v7 from Teleport|
+|status|[object](#status)|Status defines the observed state of the Teleport resource|
+
+### `status`
+
+|Field|Type|Description|
+|---|---|---|
+|teleportResourceID|integer||
+
+### `spec`
+
+|Field|Type|Description|
+|---|---|---|
+|allow|[object](#allow)|Allow is the set of conditions evaluated to grant access.|
+|deny|[object](#deny)|Deny is the set of conditions evaluated to deny access. Deny takes priority over allow.|
+|options|[object](#options)|Options is for OpenSSH options like agent forwarding.|
+
+### `options`
+
+|Field|Type|Description|
+|---|---|---|
+|cert_extensions|[][object](#cert_extensions-values)|CertExtensions specifies the key/values|
+|cert_format|string|CertificateFormat defines the format of the user certificate to allow compatibility with older versions of OpenSSH.|
+|client_idle_timeout|string|ClientIdleTimeout sets disconnect clients on idle timeout behavior, if set to 0 means do not disconnect, otherwise is set to the idle duration.|
+|create_db_user|boolean|CreateDatabaseUser enabled automatic database user creation.|
+|create_db_user_mode||CreateDatabaseUserMode allows users to be automatically created on a database when not set to off. 0 is "unspecified", 1 is "off", 2 is "keep", 3 is "best_effort_drop".|
+|create_desktop_user|boolean|CreateDesktopUser allows users to be automatically created on a Windows desktop|
+|create_host_user|boolean|CreateHostUser allows users to be automatically created on a host|
+|create_host_user_mode||CreateHostUserMode allows users to be automatically created on a host when not set to off. 0 is "unspecified"; 1 is "off"; 2 is "drop" (removed for v15 and above), 3 is "keep"; 4 is "insecure-drop".|
```

Comment:
Enum types are missing. I'm not sure what we accept here, at least the integers, maybe the string representation. @tigrato already fought with enums in CRDs